### PR TITLE
GH-13284: Add unit tests to the "RemoveUsersCmd" mmctl command

### PR DESCRIPTION
**Pull request by tomasmik**
#### Summary
Tests for the remove user from team command have been added to the mmctl tool.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13284

### CODE REVIEW

File: commands/team_test.go [hunk 1]
```diff
@@ -108,3 +108,207 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{teamArg, userArg})
+		s.Require().Equal(err, errors.New("Unable to find team '"+teamArg+"'"))
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Remove users from team, no such user", func() {
+		printer.Clean()
+		mockTeam := model.Team{Id: teamArg}
```
Comment (agnivade) [inline commands/team_test.go]: Use `mockteam := &model.Team{..}`

File: commands/team_test.go [hunk 2]
```diff
@@ -108,3 +108,207 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
 		s.Require().Len(printer.GetLines(), 0)
 	})
 }
+
+func (s *MmctlUnitTestSuite) TestRemoveUserCmd() {
+	teamArg := "example-team-id"
+	userArg := "example-user-id"
+	s.Run("Remove users from team, no args", func() {
```
Comment (agnivade) [inline commands/team_test.go]: I would suggest to give the test names as proper sentences, not short snippets. For example, for this one, I would suggest: "Remove users from team without args returns an error". This comment applies to all other tests too.

File: commands/team_test.go [hunk 3]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
 		s.Require().Len(printer.GetLines(), 0)
 	})
 }
+
+func (s *MmctlUnitTestSuite) TestRemoveUserCmd() {
+	teamArg := "example-team-id"
+	userArg := "example-user-id"
+	s.Run("Remove users from team, without args returns an error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Please remove the comma from the middle.

File: commands/team_test.go [hunk 4]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+		printer.Clean()
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{})
+		s.Require().Equal(err, errors.New("Not enough arguments."))
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Remove users from team, with one arg returns an error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Same as above, remove comma.

File: commands/team_test.go [hunk 5]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+		printer.Clean()
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{teamArg})
+		s.Require().Equal(err, errors.New("Not enough arguments."))
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Remove users from team, team not found, returns an error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Let's change it to - "Remove users from team with a non-existent team returns an error".

File: commands/team_test.go [hunk 6]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			GetTeamByName(teamArg, "").
+			Return(nil, &model.Response{Error: nil}).
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{teamArg, userArg})
+		s.Require().Equal(err, errors.New("Unable to find team '"+teamArg+"'"))
+		s.Require().Len(printer.GetLines(), 0)
+	})
+
+	s.Run("Remove users from team, user not found error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Let's change it to - "Remove users from team with a non-existent user returns an error".

File: commands/team_test.go [hunk 7]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{teamArg, mockUser.Id})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 1)
+		s.Require().Equal(printer.GetErrorLines()[0], "Can't find user '"+userArg+"'")
+	})
+
+	s.Run("Remove user by email get team by name, no error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: "Remove users from team by email and get team by name should not return an error"

File: commands/team_test.go [hunk 8]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Return(false, &model.Response{Error: nil}).
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{mockTeam.Id, mockUser.Id})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Remove users from team by email, no error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: "Remove users from team by email and get team should not return an error"

File: commands/team_test.go [hunk 9]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Return(false, &model.Response{Error: nil}).
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{mockTeam.Id, mockUser.Id})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Remove users from team by username, no error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Remove users from team by username and get team should not return an error

File: commands/team_test.go [hunk 10]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Return(false, &model.Response{Error: nil}).
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{mockTeam.Id, mockUser.Id})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Remove users from team by user no error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: Remove users from team by user and get team should not return an error

File: commands/team_test.go [hunk 11]
```diff
@@ -108,3 +108,250 @@ func (s *MmctlUnitTestSuite) TestCreateTeamCmd() {
...
+			Return(false, &model.Response{Error: nil}).
+			Times(1)
+
+		err := removeUsersCmdF(s.client, &cobra.Command{}, []string{mockTeam.Id, mockUser.Id})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetLines(), 0)
+		s.Require().Len(printer.GetErrorLines(), 0)
+	})
+
+	s.Run("Remove users from team unable to remove error", func() {
```
Comment (agnivade) [inline commands/team_test.go]: "Remove users from team with an erroneous RemoveTeamMember should return an error"

**Comment by jespino** [comment]
@tomasmik looks like the PR is already ok, can you merge master into the branch and fix the conflicts, please?

**Comment by tomasmik** [comment]
Sorry kinda busy because it's the holidays, conflicts are fixed, ready to merge.

**Pull request merged**